core/controllers/simulationcontroller/simulation_controller.go

```diff
@@ -45,6 +47,7 @@ func New(simulator simulation.Simulator) *SimulationController {
 func (c *SimulationController) Attach(router *mux.Router) {
 	r := utils.CreateSubrouter(router, c.prefix)
 	r.Methods("POST").Path("/simulate").HandlerFunc(c.DoSimulation)
+	r.Methods("DELETE").Path("/simulation/{id}").HandlerFunc(c.EndSimulation)
```

Comment:
Can we change the path parameter in the route so that it only matches integer id values?

With `gorilla/mux`, the path parameters can contain patterns like so:

```go
r.HandleFunc("/articles/{category}/{id:[0-9]+}", ArticleHandler)
```

The pattern after the colon `:` is a regex. Probably best to use an id pattern that won't match anything other than actual id values.

(from: <https://github.com/gorilla/mux#examples>)

Reply:
I think it is better if we leave test cases and create a seperate issue on jira for it

Reply:
> Can we change the path parameter in the route so that it only matches integer id values?

Ya that's okay, could you reply to the Nov 9th comment above too?

Reply:
Also, PR out of date with main branch and needs to be updated: `git rebase origin/main`

Reply:
Will do as soon as i finish the tests

Reply:
Okay sure